stats.py

```diff
@@ -1,17 +1,71 @@
 """Some simple statistical functions."""
 
-def mean(data):
-    """Takes the mean/average of a list of numbers, a simple measure of 'middle'. See: https://en.wikipedia.org/wiki/Arithmetic_mean"""
-    pass
+import numbers
+import collections
+import math
 
-def median(numbers):
-    """Finds the median or middle of a list of sorted numbers, a simple measure of 'middle'. See: https://en.wikipedia.org/wiki/Median"""
-    pass
+def mean(number_list):
+    """Takes the mean/average of a list of numbers, a simple measure of 'middle'.
+    See: https://en.wikipedia.org/wiki/Arithmetic_mean"""
+    #preconditions
+    assert isinstance(number_list, collections.Iterable), "Must be iterable."
+    for num in number_list:
+        assert isinstance(num, numbers.Number), "We need numbers yo."
+    assert len(number_list) > 1, "Please enter at least 2 numbers to be averaged."
+    #function
+    answer = (sum(number_list) / len(number_list))
+    #postconditions
+    assert isinstance(answer, numbers.Number), "Mean will return a number."
+    assert answer > min(number_list), "Mean should be higher than lowest number in list."
+    assert answer < max(number_list), "Mean should be lower than highest number in list."
+    return answer
 
-def range(numbers):
-    """Finds ths range of a list of numbers, a simple measure of variance. See: https://www.mathsisfun.com/definitions/range-statistics-.html"""
-    pass
+def median(number_list):
+    """Finds the median or middle of a list of sorted numbers, a simple measure of 'middle'.
+    See: https://en.wikipedia.org/wiki/Median"""
+    #preconditions
+    assert isinstance(number_list, collections.Iterable), "Must be iterable."
+    for num in number_list:
+        assert isinstance(num, numbers.Number), "We need numbers yo."
+    assert len(number_list) > 2, "Please enter at least 3 numbers to calculate median."
```

Comment:
this is too strict. you can calculate the median of a single element list as well as a two element list. 

the median of an even number of elements is the average of it's two most center elements

https://en.wikipedia.org/wiki/Median

i'd add a test of a multiple element list:

[1, 2, 3, 4, 5, 6]

the median of that list is 3 + 4 / 7 = 3.5

once your test passes that you know you're golden